props/saber_BC_buttons.h

```diff
@@ -2007,6 +2045,55 @@ void DoSavedTwist() {
 
 #endif  // BC_DUAL_BLADES
 
+  bool use_fake_id_ = false;
+  float fake_id_ = 0;
+  size_t best_config_before_faking_ = SIZE_MAX;
+  bool play_no_change_sound_ = false;
+
+  float id(bool announce = false) override {
+    if (use_fake_id_) {
+      float real_id = PropBase::id(announce);
+      size_t real_best_config = FindBestConfigForId(real_id);
+
+      if (real_best_config != best_config_before_faking_) {
+        // Real blade has changed, exit manual mode
+        use_fake_id_ = false;
+        if (current_config - blades == real_best_config) {
+          play_no_change_sound_ = true;
+        }
+        return real_id;
+      }
+      return fake_id_;
+    }
+
+    return PropBase::id(announce);
+  }
+
+  void TriggerBladeID() {
+    use_fake_id_ = false;
+    FindBladeAgain();
+    PlayArraySound();
+  }
+
+  void NextBladeArray() {
+    best_config_before_faking_ = FindBestConfigForId(PropBase::id(false));
```

Comment:
I realized that this doesn't work if the blade is on.
Your key bindings are all for MODE_OFF, so maybe it doesn't matter, but the blade can still be on if people have idle effects, long fadeouts, or battery effects.

I think we can replace it with:
```
 if (!use_fake_id_) best_config_before_faking_ = current_config - blades;
```

Reply:
Great. In my limited testing, this works the same, so hopefully it helps other scenarios as well.
Edited.